operator/controller/src/main/java/io/apicurio/registry/operator/resource/ResourceFactory.java

```diff
@@ -28,19 +37,128 @@ public class ResourceFactory {
+                "quay.io/apicurio/apicurio-registry-ui:latest-snapshot",
+                List.of(new ContainerPortBuilder().withName("http").withProtocol("TCP").withContainerPort(8080).build()),
+                new ProbeBuilder().withHttpGet(new HTTPGetActionBuilder().withPath("/config.js").withPort(new IntOrString(8080)).withScheme("HTTP").build()).build(),
+                new ProbeBuilder().withHttpGet(new HTTPGetActionBuilder().withPath("/config.js").withPort(new IntOrString(8080)).withScheme("HTTP").build()).build(),
+                Map.of("cpu", new Quantity("100m"), "memory", new Quantity("256Mi")),
+                Map.of("cpu", new Quantity("200m"), "memory", new Quantity("512Mi"))
+                // spotless:on
+        );
+        addDefaultLabels(r.getMetadata().getLabels(), primary, COMPONENT_UI);
         addSelectorLabels(r.getSpec().getSelector().getMatchLabels(), primary, COMPONENT_UI);
+        addDefaultLabels(r.getSpec().getTemplate().getMetadata().getLabels(), primary, COMPONENT_UI);
         return r;
     }
 
+    /**
+     * Merge default values for a Deployment into the target PTS (from spec).
+     */
+    private static void mergeDeploymentPodTemplateSpec(
+            // spotless:off
+            PodTemplateSpec target,
+            String containerName,
+            String image,
+            List<ContainerPort> ports,
+            Probe readinessProbe,
+            Probe livenessProbe,
+            Map<String, Quantity> requests,
+            Map<String, Quantity> limits
+            // spotless:on
+    ) {
+        if (target.getMetadata() == null) {
+            target.setMetadata(new ObjectMeta());
+        }
+        var c = getContainerFromPodTemplateSpec(target, containerName);
+        if (c == null) {
+            if (target.getSpec() == null) {
+                target.setSpec(new PodSpec());
+            }
+            c = new Container();
+            c.setName(containerName);
+            if (target.getSpec().getContainers() == null) {
+                target.getSpec().setContainers(new ArrayList<>());
+            }
+            target.getSpec().getContainers().add(c);
+        }
+        if (isBlank(c.getImage())) {
+            c.setImage(image);
+        }
+        if (c.getEnv() != null && !c.getEnv().isEmpty()) {
+            throw new OperatorException("""
+                    Field spec.(app/ui).podTemplateSpec.spec.containers[name = %s].env must be empty. \
+                    Use spec.(app/ui).env to configure environment variables.""".formatted(containerName));
+        }
+        if (c.getPorts() == null) {
+            c.setPorts(new ArrayList<>());
+        }
+        mergeNotOverride(c.getPorts(), ports, ContainerPort::getName);
+        if (c.getReadinessProbe() == null) {
+            c.setReadinessProbe(readinessProbe);
+        }
+        if (c.getLivenessProbe() == null) {
+            c.setLivenessProbe(livenessProbe);
+        }
+        if (c.getResources() == null) {
```

Comment:
After a bit of thinking about it, I think the logic here should be:
if `resources` is defined in the `PodTemplateSpec` it fully overrides the defaults.
This allows people to remove the limits when necessary.

Reply:
Done.